src/linter/ui5Types/SourceFileLinter.ts

```diff
@@ -120,12 +124,154 @@ export default class SourceFileLinter {
 				context: this.#context,
 				checker: this.#checker,
 			});
+		} else if (ts.isPropertyAssignment(node) && this.isPropertyInMetadata(node)) {
+			this.analyzeMetadataProperty(node.name.getText(), node);
```

Comment:
Do I understand this right that we run the `this.isPropertyInMetadata(node)` check for every single property within the MO metadata (i.e. multiple times for the same class)?

I would hope that this can be solved differently, so that we only need to check for this once. Maybe the entry for the check can be done at the ClassDeclaration and then we know that all properties below that node are part of the class. As we plan other checks as well like checking for renderer of controls (which are also sub-classes of ManagedObject), we should try to only check for the inheritance chain at a single place and then run checks based on the class type.

Reply:
Indeed, I was wondering what is the best approach to do so.
That's why I put those pef measurements. Basically, for every prop, we check whether it's a `metadata` child and if the control is child of ManagedObject.

The tricky part here is that nodes are visited recursivelly in depth-first approach. And I'm not sure how to handle the check that the current node is actually inside the `metadata` property.

Reply:
These are the performance results for the following check:
```js
if (ts.isPropertyAssignment(node) && this.isPropertyInMetadata(node)) {
```

### Base- no check at all
| Command | Mean [s] | Min [s] | Max [s] | Relative |
|:---|---:|---:|---:|---:|
| `node /Users/i326076/SAPDevelop/ui5-linter/bin/ui5lint.js` | 39.019 ± 0.422 | 38.311 | 39.921 | 1.00 |


### Check- just the check, `this.analyzeMetadataProperty(node.name.getText(), node);` is not executed.
| Command | Mean [s] | Min [s] | Max [s] | Relative |
|:---|---:|---:|---:|---:|
| `node /Users/i326076/SAPDevelop/ui5-linter/bin/ui5lint.js` | 39.006 ± 0.229 | 38.657 | 39.338 | 1.00 |

Reply:
This is the alternative implementation. Still polishing it, but I have doubts it will be quicker as it will be hard to optimise the traversal, so metdata nodes to be visited only once: https://github.com/SAP/ui5-linter/pull/360

Reply:
WIth https://github.com/SAP/ui5-linter/pull/360 merged in this branch the measurements are alike the main branch